### PR TITLE
Reset scroll position after navigation. Fixes #10482

### LINKED ISSUE #10482: [Blazor] Scroll position persists during page navigation.
### Describe the bug
When navigating from page to page, the scroll position is persisted instead of resetting.

### To Reproduce
1. Create a razor page and call it `Page1.razor`, then paste the following code on the page:

```
@for (int i = 0; i <= 1000; i++)
{
    <h1>I will pay attention in school</h1>
}
<a href="Page2">Go to page 2</a>
```

2. Create a second razor page and call it `Page2.razor`, then paste the following code on the page:

```
@for (int i = 0; i <= 1000; i++)
{
    <h1>1 + 1 = 3</h1>
}
```
3. Start the app, navigate to `Page1 `and scroll all the way down until you see the `Go to page 2` link.
4. Click the `Go to page 2` link.

The correct behavior (in my opinion) should be for Page2 scroll position to be at the top of the page. Instead, Page2 is scrolled down to the same scroll position Page1 was.

This happens at least in Chrome and Edge.

### PR DESCRIPTION
**Pull request by SteveSandersonMS**
This only affects link clicks and programmatic navigation.

For back/forwards, browsers already apply their own native scroll position restoration behavior, which varies between browsers, so we leave that alone.

### CODE REVIEW

File: src/Components/test/E2ETest/Tests/RoutingTest.cs [hunk 1]
```diff
@@ -418,6 +418,40 @@ public void RoutingToComponentOutsideMainAppDoesNotWork()
...
+            Browser.Equal("This is a long page you can scroll.", () => app.FindElement(By.Id("test-info")).Text);
+            BrowserScrollY = 500;
+            Browser.True(() => BrowserScrollY > 300); // Exact position doesn't matter
+
+            app.FindElement(By.Id("go-to-longpage2")).Click();
+            Browser.Equal("This is another long page you can scroll.", () => app.FindElement(By.Id("test-info")).Text);
+            Browser.Equal(0, () => BrowserScrollY);
+        }
+
+        private long BrowserScrollY
```
Comment (javiercn) [inline src/Components/test/E2ETest/Tests/RoutingTest.cs]: This is kind of confusing. I would have simply used methods for this.

**Review by javiercn — APPROVED** [review]
Great change!

Straightforward and with E2E, nice!

**Comment by kaveh6202** [comment]
I think there is a problem with this approach , you see i have a collapse component in my page and it is kind of in the bottom of the page . when i toggle the tab , the normal behavior i think is page does not scroll and stay put since if have inner page link ( starts with # ) in my <a> tag .

**Comment by smirceamihai** [comment]
Agree with @kaveh6202
Please consider:
- Turning this into an optional (parameter) feature for the NavigateTo method
- Disabling the scroll reset if the navigation goes to the same LocalPath (only query params change)

For us the workaround was overriding the default behavior with an interop method:

`async function navigateToState (url) 
{
        history.replaceState(null, null, url); 
        await DotNet.invokeMethodAsync(
            'Microsoft.AspNetCore.Components.WebAssembly',
            'NotifyLocationChanged',
            url,
            false // IsNavigationIntercepted
        );
}`

**Comment by christallire** [comment]
> Agree with @kaveh6202 Please consider:
> 
> * Turning this into an optional (parameter) feature for the NavigateTo method
> * Disabling the scroll reset if the navigation goes to the same LocalPath (only query params change)
> 
> For us the workaround was overriding the default behavior with an interop method:
> 
> `async function navigateToState (url) { history.replaceState(null, null, url); await DotNet.invokeMethodAsync( 'Microsoft.AspNetCore.Components.WebAssembly', 'NotifyLocationChanged', url, false // IsNavigationIntercepted ); }`

for anyone who stumbled this on .NET 7
there's some internal changes at the API. need to add `history` argument after the url like this:
```
await DotNet.invokeMethodAsync( 'Microsoft.AspNetCore.Components.WebAssembly', 'NotifyLocationChanged', url, '', false);
```

**Pull request merged**